### ISSUE TITLE
Mobile Launch

### ISSUE THREAD

**Issue by benjamin658** [thirdgen88/ignition-docker#4]
Hi, I'm trying to use this image with Ignition version 7.9.10
everything is working great, except the mobile launch

here is the error log:
```
java.net.SocketException: Connection reset

at java.net.SocketInputStream.read(SocketInputStream.java:210)

at java.net.SocketInputStream.read(SocketInputStream.java:141)

at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)

at java.io.BufferedInputStream.read(BufferedInputStream.java:265)

at java.io.DataInputStream.readShort(DataInputStream.java:312)

at com.inductiveautomation.mobile.gateway.MobileVMManager$ClientVMImpl.getDirtyRectangles(MobileVMManager.java:511)

at com.inductiveautomation.mobile.gateway.servlets.MobileSession.getDirtyRectangles(MobileSession.java:336)

at com.inductiveautomation.mobile.gateway.servlets.MobileDataServlet$RectsBase64.handle(MobileDataServlet.java:358)

at com.inductiveautomation.mobile.gateway.servlets.MobileDataServlet.doRequest(MobileDataServlet.java:203)

at com.inductiveautomation.mobile.gateway.servlets.MobileDataServlet.doGet(MobileDataServlet.java:144)

at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)

at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)

at com.inductiveautomation.ignition.gateway.bootstrap.MapServlet.service(MapServlet.java:85)

at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:837)

at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:583)

at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)

at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)

at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226)

at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1160)

at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:511)

at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)

at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1092)

at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)

at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:52)

at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)

at org.eclipse.jetty.server.Server.handle(Server.java:518)

at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:308)

at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:244)

at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:273)

at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95)

at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:186)

at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:273)

at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95)

at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93)

at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceAndRun(ExecuteProduceConsume.java:246)

at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:156)

at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:654)

at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:572)

at java.lang.Thread.run(Thread.java:748)
```
just want to make sure that if there is anyone has launched mobile successfully?

**Comment by thirdgen88** [comment]
Interesting..  I've not tried the mobile functionality yet under the Docker container..  I will try this weekend and see if I can reproduce..  I don't think there is anything that should keep it from running, as we're still using Java 8 for the 7.9 image (which is needed for the Mobile module to function properly).

**Comment by M0rfic** [comment]
I second that and also I'd like to add for reference https://support.inductiveautomation.com/index.php?/Knowledgebase/Article/View/33/0/running-the-mobile-module-on-linux. I've been trying to find some time to investigate before opening an issue, maybe dig a bit deeper and come up with a more detailed solution, but so far was too busy, sry about that

**Comment by thirdgen88** [comment]
>I second that and also I'd like to add for reference https://support.inductiveautomation.com/index.php?/Knowledgebase/Article/View/33/0/running-the-mobile-module-on-linux. I've been trying to find some time to investigate before opening an issue, maybe dig a bit deeper and come up with a more detailed solution, but so far was too busy, sry about that

@M0rfic thank you for sharing that, I had forgotten all about that (despite reading it long ago)!  Looks like I will need to add a few more dependencies for that to work then.  I will check into what it will take to get it integrated!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by thirdgen88** [comment]
Okay, I've worked up what I think is a pretty good solution to this!  I created a new dedicated image, [kcollins/xvfb](https://hub.docker.com/r/kcollins/xvfb), that can be used as a sidecar for the Ignition gateway.  It works by sharing the X11 socket through a named volume that you attach to both the Ignition gateway and Xvfd containers.

I've updated the [ignition-examples](https://github.com/thirdgen88/ignition-examples) repository as well with an [ignition-mobile](https://github.com/thirdgen88/ignition-examples/tree/master/ignition-mobile) section that has a simple docker-compose.yml example.

Let me know if this works for you, @benjamin658.
